### ISSUE TITLE
build/serve generates wrong scripts.js URL when index.html non-standard

### ISSUE THREAD

**Issue by benb7760** [angular/angular-cli#7973]
### Bug Report or Feature Request (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
```

### Versions.
ng version
```
@angular/cli: 1.4.5
node: 6.11.0
os: win32 x64
@angular/animations: 4.4.4
@angular/common: 4.4.4
@angular/compiler: 4.4.4
@angular/core: 4.4.4
@angular/forms: 4.4.4
@angular/http: 4.4.4
@angular/platform-browser: 4.4.4
@angular/platform-browser-dynamic: 4.4.4
@angular/router: 4.4.4
@angular/cli: 1.4.5
@angular/compiler-cli: 4.4.4
@angular/language-service: 4.4.4
typescript: 2.3.4
```

### Repro steps.
The below steps may sound strange - so for context, I've made a project which contains two apps, but they share some files from the directory above (shared index.html, assets, scripts). This saves me having to maintain two equal copies of my assets/index.html

1. Generate project
2. move the contents of /src/ down one level, to /src/application
3. move index.html back up to /src/index.html
4. in .angular-cli.json, change root from `src` to`src/application`
5. in .angular-cli.json, change index from `index.html` to `../index.html`
6. in .angular-cli.json, add some random JS file from `../../node_modules/*` (doesn't really matter which one is used for purposes of repro
7. run ng serve
8. browse to page. The URL for the vendor bundles, etc, are correct, but the scripts.js is incorrect and cannot be found. it seems to have included an extra path in it. the issue can also be seen after running `ng build`, the index.html file clearly has a different URL:

```
<script type="text/javascript" src="dist/inline.bundle.js"></script>
<script type="text/javascript" src="dist/polyfills.bundle.js"></script>
<script type="text/javascript" src="dist/application\scripts.bundle.js"></script>
<script type="text/javascript" src="dist/styles.bundle.js"></script>
<script type="text/javascript" src="dist/vendor.bundle.js"></script>
<script type="text/javascript" src="dist/main.bundle.js"></script></body>
```

### Desired functionality.
The URL output for the scripts bundle should use the same logic/create the same output format to the vendor/styles/etc bundles

### Mention any other details that might be useful.
If required I can create a github repo that reproduces this

**Comment by hansl** [comment]
Hi @benb7760 

Could you follow the template and run `ng version` and report the result back here? Also, having a repro could be useful.

Thanks.

**Comment by benb7760** [comment]
@hansl I hope this is easy enough to follow:

I have put together a repro [here](https://github.com/benb7760/angular-testing/commits/cli-7973) and provided two key commits. 

Commit 1: [7cdf527](https://github.com/benb7760/angular-testing/commit/7cdf527b5d83ae1ce963e603f759697fcd3c0555)
This shows an ordinary setup. When I build, the bundles and the index.html are found in the dist directory.

Commit 2: [6adf3c9](https://github.com/benb7760/angular-testing/commit/6adf3c9289f9255e7fdd419dd67b44048e8b30d2)
I have configured to 'share' files index.html, favicon.ico, assets, polyfills.ts, styles.css.
I have configured the outDir to 'dist/application1'
When I build:
* index.html & favicon.ico go to 'dist' - should this not be 'root/application1'?
* index.html references the script bundle from the wrong path
* Inside 'dist/application1', I see all files as expected (minus index/favico)

Please let me know if there is anything else I can provide

ng version results:

```
@angular/cli: 1.4.5
node: 6.11.0
os: win32 x64
@angular/animations: 4.4.4
@angular/common: 4.4.4
@angular/compiler: 4.4.4
@angular/core: 4.4.4
@angular/forms: 4.4.4
@angular/http: 4.4.4
@angular/platform-browser: 4.4.4
@angular/platform-browser-dynamic: 4.4.4
@angular/router: 4.4.4
@angular/cli: 1.4.5
@angular/compiler-cli: 4.4.4
@angular/language-service: 4.4.4
typescript: 2.3.4
```

**Comment by benb7760** [comment]
It appears it's related to the root directory. If I move this up one level and adjust the paths, it works fine.

Should this be the solution, or is there still a bug to fix?

**Comment by clydin** [comment]
This should be corrected in 1.6.3.  If the problem still occurs, please re-open the issue.
